adapters/repos/db/replication.go

```diff
@@ -253,12 +254,34 @@ func (i *Index) IncomingCreateShard(ctx context.Context, className string, shard
 func (i *Index) IncomingReinitShard(ctx context.Context,
 	shardName string,
 ) error {
-	shard, err := i.getOrInitLocalShard(ctx, shardName)
-	if err != nil {
-		return fmt.Errorf("shard %q does not exist locally", shardName)
+	shard := func() ShardLike {
```

Comment:
question(non-blocker): I don't understand why we need a lambda here. I think it would simply the code to simply assign to shard explicitely.

Reply:
If it's only for the lock defer i'd say we can explicitely unlock *or* use code blocks with a pair of `{}` IMO

Reply:
yeah i see your point, given this part wasn't changed by me but i think it's just to scope the GuardLock, IIRC we have some function actually does that , we could use later on. it could be applied in refactor patch though to not start the whole pipeline

Reply:
great, forgot about `{}` block.
I prefer using `defer` as it is always called, thus being safer then explicit unlock calls. We had deadlock issues before due to mutexes not being unlocked in case of rather harmless panics.

Reply:
actually block does not work. defer is called at the end of method containing block. So either anonymous function needs to be used or `KeyLocker` would need to have method accepting callback to be executed while locked.